internal/services/network/network_security_group_resource.go

```diff
@@ -57,7 +57,6 @@ func resourceNetworkSecurityGroup() *pluginsdk.Resource {
 				Type:       pluginsdk.TypeSet,
 				ConfigMode: pluginsdk.SchemaConfigModeAttr,
 				Optional:   true,
-				Computed:   true,
```

Comment:
hey @catriona-m 

Network Security Groups and Rules are one of a number of resources within the Provider which support defining the nested item (in this case Network Security Rules) both within the parent resource (to say, "only these rules should exist") and externally as a separate resource (to say, "just add another rule, I'm not concerned about which ones exist"). This behaviour exists for a handful of resources within the Provider, notably Virtual Networks and Subnets, but there's a number of others too - in each case these are called out in the documentation.

This is implemented by making the nested item `Computed` within the parent resource - however in a number of instances to both maintain compatibility with user configs and handle the API behaviour here, we're also taking advantage of `ConfigModeAttr` which allows assigning a list of items as an attribute, rather than as a block (as can be seen here).

Whilst we'll look to remove both `ConfigModeAttr` and the `Computed` attribute here in (likely) v4.0 of the Provider, unfortunately this isn't something we can do right now as it'd be quite a large breaking change for a number of reasons. Firstly, the Virtual Network resource itself uses a combined Create/Update right now, meaning that removing the `Computed: true` would highlight a diff for users, which ultimately wouldn't work due to the way these are parsed out of the config - whilst removing `ConfigModeAttr` would be a breaking configuration change for users.

As mentioned above this is likely something we'll be looking to fix in v4.0 of the Provider across most resources, splitting the Create and Update methods such that `ignore_changes` can work in most fields - and in these cases likely removing `ConfigModeAttr` too. We'll also look to extend the `subnet` block within the `azurerm_virtual_network` resource at this time to include additional fields too, since this'd (potentially) require changes to user configs.

However unfortunately this isn't something we can ship at this time - and as such whilst I'd like to thank you for this contribution I'm going to close this PR for the moment.

Thanks!